pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -412,6 +413,60 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func TestPriorityQueue_Activate(t *testing.T) {
+	var objs []runtime.Object
+	q := NewTestQueueWithObjects(context.Background(), newDefaultQueueSort(), objs)
+	qPodInfo := framework.QueuedPodInfo{
+		PodInfo: highPriNominatedPodInfo,
+	}
+
+	t.Run("test PriorityQueue.activate for pod in activeQ", func(t *testing.T) {
+		// No activated pod
+		assert.Equal(t, 0, q.activeQ.Len())
+
+		// Add one to activeQ
+		q.activeQ.Add(&qPodInfo)
+		defer q.activeQ.Delete(&qPodInfo)
+
+		// Should be false on PriorityQueue.activate so still only one activated pod
+		q.Activate(map[string]*v1.Pod{"test_pod": highPriNominatedPodInfo.Pod})
+		assert.Equal(t, 1, q.activeQ.Len())
+	})
+
+	t.Run("test PriorityQueue.activate for pod not in unschedulableQ", func(t *testing.T) {
+		// No activated pod
+		assert.Equal(t, 0, q.activeQ.Len())
+
+		// Could not activate since highPriNominatedPod not in unschedulableQ
+		q.Activate(map[string]*v1.Pod{"test_pod": highPriNominatedPodInfo.Pod})
+		assert.Equal(t, 0, q.activeQ.Len())
+	})
+
+	t.Run("test PriorityQueue.activate for pod in unschedulableQ", func(t *testing.T) {
+		defer q.activeQ.Delete(&qPodInfo)
```

Comment:
maybe you should also check the length of unschedulableQ

Reply:
PTAL